### PR TITLE
Changed the sensor graph and weather displays to work with the new query interface for the new katstore.

### PR DESCRIPTION
**Pull request by fjoubert**
Also:
- Added a 'Search Old Sensors' checkbox to toggle searching for stale sensors (off by default).

**IMPORTANT:**
- This PR will NOT bring these changes to the production version. For instructions on how to build the production version, read the [README](https://github.com/ska-sa/katgui/blob/master/README.md).
- The utilisation report display still requires reworking when the endpoint for calculating the duration of a sensor's values for a given time window exists.
- This PR must only be merged (and a new version built) when the new katstore query interface is up and running.
- This PR expects that the katstore endpoint on the CAM portal node will redirect (by way of an nginx reverse proxy) to the correct katstore query interface, which is hosted on a non-CAM control node.
- When the query interface is ready and this PR is to be merged, then the katstore-webserver can be completely deprecated from [katcamconfig](https://github.com/ska-sa/katcamconfig/blob/master/systems/base_cam_processes.conf#L319-L323) and from [katportal](https://github.com/ska-sa/katportal/tree/master/katportal/katstore_webserver).

### CODE REVIEW

File: app/reports/utilisation-report.js [hunk 1]
```diff
@@ -509,13 +508,13 @@
                 DataService.sampleValueDuration(vm.interlockReportSensorsRegex, startDate, endDate).then(function (result) {
                     vm.creatingInterlockReport = false;
                     if (result.data) {
```
Comment (ajoubertza) [inline app/reports/utilisation-report.js]: Should this one also be `result.data.data`?

File: app/sensor-graph/sensor-graph.js [hunk 2]
```diff
@@ -387,42 +401,16 @@
...
-                if (newData.length > 0) {
-                    vm.redrawChart(newData, hasMinMax);
-                }
-            } else if (subject.startsWith('sensor.')) {
+            if (subject.startsWith('sensor.')) {
                 if (angular.isDefined(_.findWhere(vm.sensorNames, {name: message.name}))) {
                     vm.redrawChart([{
                         sensor: message.name,
                         value_ts: message.value_ts * 1000,
                         sample_ts: message.time * 1000,
```
Comment (ajoubertza) [inline app/sensor-graph/sensor-graph.js]: Oh, I see here we multiple message time with factor 1000 (guessing this is a NATS message).  We could change the keys to `value_time` and `sample_time` to match katstore responses.

File: app/weather/weather.js [hunk 3]
```diff
@@ -80,7 +80,7 @@
             });
 
         vm.initSensors = function() {
-            var startDate = vm.getTimestampFromHistoricalRange();
+            var startDate = vm.getTimestampFromHistoricalRange() / 1000;
```
Comment (ajoubertza) [inline app/weather/weather.js]: It is really confusing that Javascript uses time in milliseconds, and the rest of CAM uses seconds.  Impossible to infer units from variable names.  We could add a suffix, e.g. `startDateMs` or `startDateSec`.

File: app/weather/weather.js [hunk 4]
```diff
@@ -115,12 +123,12 @@
...
-                var sensorName = sensor[4];
+                var sensorName = sensor.sensor;
                 var latestSensor = {
-                    status: sensor[5],
+                    status: sensor.status,
                     sensor: sensorName,
-                    value: sensor[3],
-                    sample_ts: sensor[0],
+                    value: sensor.avg_value,
+                    sample_ts: sensor.min_sample_time * 1000,
```
Comment (ajoubertza) [inline app/weather/weather.js]: Another place we might want to standardise on the time variable naming.  `sample_time` or `sample_ts`

File: app/weather/weather.js [hunk 5]
```diff
@@ -80,8 +80,8 @@
             });
 
         vm.initSensors = function() {
-            var startDate = vm.getTimestampFromHistoricalRange() / 1000;
-            vm.dataTimeWindow = new Date().getTime() - startDate;
+            var startDateSec = vm.getTimestampFromHistoricalRange() / 1000;
+            vm.dataTimeWindow = new Date().getTime() - startDateSec;
```
Comment (ajoubertza) [inline app/weather/weather.js]: `Date().getTime()` seems to return a time in milliseconds, so don't combine with a time in seconds.

File: dist/version.txt [hunk 6]
```diff
@@ -1 +1 @@
-{"version": "1.0.1", "buildDate": "1540390246451"}
+{"version": "1.0.1", "buildDate": "1540398489730"}
```
Comment (ajoubertza) [inline dist/version.txt]: @ofaletse Were the dist files rebuilt after merging in master?  Please remove old minified files - there are some extras now.

**Comment by ofaletse** [comment]
@ajoubertza 
I thought I have done that, but It looks like other commit has overridden that.

**Comment by ajoubertza** [comment]
I want to move to a properly named feature branch for this work:  [`feature/new-katstore-integration`](https://github.com/ska-sa/katdeploy/tree/feature/new-katstore-integration)
That feature branch is currently pointing to the same commit as the head of this old user branch, so can I close this super long PR, and delete the old branch?  The changes from the feature branch will be merged to master when we are ready.
@ofaletse @bxaia or anyone else have a comment?

**Comment by ajoubertza** [comment]
Closing this PR, as the branch will be deleted - using feature branch instead.

**Pull request closed without merging**